src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.Accessibility.cs

```diff
@@ -0,0 +1,392 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Symbols
+{
+    public sealed partial class ModuleInitializersTests
+    {
+        [Fact]
+        public void VisibilityMustNotBePrivate()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer]
+    private static void M() { }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics(
```

Comment:
>VerifyDiagnostics [](start = 24, length = 17)

I think we we should be using VerifyEmitDiagnostics for this feature to make sure that, if we attempt to emit in presence of errors, the code we added to CSharpCompilation is not going to cause problems. This applies to all tests for this feature.